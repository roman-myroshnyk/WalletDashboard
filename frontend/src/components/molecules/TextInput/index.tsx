import { FC } from 'react';
import Input from '@/atoms/Input/index';

export interface IProps {
  value: string;
  placeholder:string;
  disabled:boolean;
  onSubmit: () => void;
  onChange: (value:string) => void;
  testid?:string;
}

const TextInput:FC<IProps> = ({
  value,
  placeholder,
  disabled,
  onSubmit,
  onChange,
  testid,
}) => (
  <Input
    value={value}
    placeholder={placeholder}
    disabled={disabled}
    onSubmit={onSubmit}
    onChange={onChange}
    multiLine={false}
    inputType="search"
    keyboard="search"
    truncate
    testid={testid}
  />
);

TextInput.defaultProps = {
  testid: 'text input',
};

export default TextInput;
