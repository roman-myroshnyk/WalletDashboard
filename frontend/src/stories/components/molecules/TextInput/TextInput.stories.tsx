import { Meta, Story } from '@storybook/react/types-6-0';
import TextInput, { IProps } from '@/molecules/TextInput';

export default {
  title: 'molecules/TextInput',
  component: TextInput,
  argTypes: {
    value: {
      controls: 'text',
    },
    placeholder: {
      controls: 'text',
    },
    disabled: {
      controls: 'boolean',
    },
    onSubmit: {
      action: 'onSubmit: enter pressed',
    },
    onChange: {
      action: 'onChange: input changed',
    },
  },
} as Meta;

const Template:Story<IProps> = (args) => (
  <TextInput {...args} />
);

export const Default = Template.bind({});

Default.args = {
  value: '',
  placeholder: 'I am placeholder',
  disabled: false,
};
