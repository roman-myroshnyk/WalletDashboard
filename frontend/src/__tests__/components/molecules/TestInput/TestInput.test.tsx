import '.../../../mocks/framer-motion';
import { render, fireEvent } from '@testing-library/react';
import TextInput from '@/molecules/TextInput';

describe('molecules/TextInput', () => {
  it('TextInput should render', () => {
    const handleSubmit = jest.fn();
    const handleChange = jest.fn();

    const { container } = render(
      <TextInput
        value="I am text input"
        placeholder="I am placeholder"
        disabled={false}
        onSubmit={handleSubmit}
        onChange={handleChange}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('TextInput should render disabled state', () => {
    const handleSubmit = jest.fn();
    const handleChange = jest.fn();

    const { container } = render(
      <TextInput
        value="I am text input"
        placeholder="I am placeholder"
        disabled
        onSubmit={handleSubmit}
        onChange={handleChange}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('TextInput should handle change', () => {
    const handleSubmit = jest.fn();
    const handleChange = jest.fn();

    const { getByTestId } = render(
      <TextInput
        value="I am text input"
        placeholder="I am placeholder"
        disabled={false}
        onSubmit={handleSubmit}
        onChange={handleChange}
        testid="text input"
      />,
    );
    const input = getByTestId('text input');
    fireEvent.change(input, {
      target: {
        value: 'a',
      },
    });
    expect(handleChange).toBeCalledTimes(1);
  });
  it('TextInput should handle Submit', () => {
    const handleSubmit = jest.fn();
    const handleChange = jest.fn();

    const { getByTestId } = render(
      <TextInput
        value="I am text input"
        placeholder="I am placeholder"
        disabled={false}
        onSubmit={handleSubmit}
        onChange={handleChange}
        testid="text input"
      />,
    );
    const input = getByTestId('text input');
    fireEvent.keyDown(input, {
      key: 'Enter',
    });
    expect(handleSubmit).toBeCalledTimes(1);
  });
});
