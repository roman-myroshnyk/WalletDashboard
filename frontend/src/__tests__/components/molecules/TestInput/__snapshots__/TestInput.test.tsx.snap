// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/TextInput TextInput should render 1`] = `
<div>
  <input
    class="tag tagTruncate"
    data-testid="text input"
    id=""
    inputmode="search"
    min="1"
    placeholder="I am placeholder"
    rows="1"
    step="0.5"
    style="text-align: left;"
    type="search"
    value="I am text input"
  />
</div>
`;

exports[`molecules/TextInput TextInput should render disabled state 1`] = `
<div>
  <input
    class="tag tagTruncate"
    data-testid="text input"
    disabled=""
    id=""
    inputmode="search"
    min="1"
    placeholder="I am placeholder"
    rows="1"
    step="0.5"
    style="text-align: left;"
    type="search"
    value="I am text input"
  />
</div>
`;
